src/Components/Scorecard.js

```diff
@@ -1,150 +1,114 @@
+    return scorecard.map(player => {
       return (
         <Player
           name={player.name}
           key={player.id}
           id={player.id}
-          par={scorecard.layout.holes.find(hole => {
+          par={courseInfo.currentCourse.layout.holes.find(hole => {
             return hole.hole_number === currentHole.number
           }).par}
-          totalScore={currentHole.players[`${player.id}`].totalScore}
+          totalScore={findTotalScore(player.id)}
           changeScore={changeScore}
         />
       )
     })
   }
 
-  const nextHole = (prevHoleScores) => {
-    const hole = scorecard.layout.holes.find(hole => hole.hole_number === currentHole.number + 1)
-    if(hole.hole_number === scorecard.layout.holes.length){
+  const findTotalScore = (id) => {
+    let playerScores = scores.filter(score => score.player_id === id) 
+      return playerScores.length > 0 ? playerScores.reduce((acc, player) => {
+        acc += player.score
+        return acc
+      }, 0) : 0
+  }
+
+  const checkLastHole = () => {
+    return inProgress ?
+      <button className='next-hole-btn' onClick={postHoleScores}>NEXT HOLE</button> :
+      <button className='end-round-btn' onClick={postHoleScores}>END ROUND</button>
+  }
+
+  const changeScore = (id, score) => {
+    let player = scorecard.find(player => player.id === id)
+
+    player.score = score
+    const otherPlayers = scorecard.filter(player => player.id !== id)
+    setScorecard([...otherPlayers, player])
+    setCurrentHole(currentHole)
+  }
+
+  const nextHole = () => {
+    const hole = courseInfo.currentCourse.layout.holes.find(hole => hole.hole_number === currentHole.number + 1)
+    if(hole.hole_number === courseInfo.currentCourse.layout.holes.length){
       setInProgress(false)
     }
     setCurrentHole({
       number: hole.hole_number,
-      players: updatePlayerScores(prevHoleScores),
       par: hole.par,
       distance: hole.distance
     })
   }
 
-  const updatePlayerScores = (prevHoleScores) => {
-    const updatedPlayers = {}
-    prevHoleScores.forEach(player => {
-      updatedPlayers[player.player_id] = {
-        name: currentHole.players[player.player_id].name,
-        id: currentHole.players[player.player_id].id,
-        score: 0,
-        totalScore: player.score + currentHole.players[player.player_id].totalScore
-      }
-    })
-    return updatedPlayers
-  }
-
   const postHoleScores = () => {
-    const currentPlayers = Object.keys(currentHole.players)
-    const playerScores = currentPlayers.map(player_id => {
-      return {
-        player_id: currentHole.players[player_id].id,
-        strokes: currentHole.players[player_id].score
-      }
-    })
-
     const body = {
       hole: {
         hole_number: currentHole.number,
-        player_scores: playerScores
+        player_scores: scorecard.map(player => {
+          return {
+            player_id: player.id,
+            strokes: player.score === 0 ? 
+              currentHole.par : 
```

Comment:
This makes sure that if the player doesn't press a button, the par for the hole will be used.